### DISCUSSION TITLE
Exporting flip-chip gds

### DISCUSSION THREAD

**Discussion by Qvadi** [iqm-finland/KQCircuits#81] [Q&A]
Hi KQCircuits,
When saving/exporting the files of a multi-chip design, how do I mirror image the top chip layers. When I created the PCell for the chip, the top chip_frame was mirrored (this is a parameter that can be set). I know that there is this option for mirroring for a mask export. But how do I do this, if I want to just export the multi_chip gds? I tried looking for a parameter to do this but couldnt find any.

**Comment by qpavsmi** [comment]
Hi, sorry for the delay. By default we have `Chip` set up that if you set `frames_enabled` parameter to `[0,1]`, then the chip will have a chip frame for the bottom "1t1" face and top "2b1" face. The labels of the top face will be mirrored. However the geometry of the "2b1" face is placed as is, without mirroring. So given some pya.Region called `region`, if you add the code `self.cell.shapes(self.get_layer("base_metal_gap_wo_grid", face_id=1)).insert(region)`, you will have the region placed on the top face.

Indeed, during mask export, since we separate the OAS files based on chip faces, we have the entire wafer for "2b1" face in its own file and already mirrored, so that the labels are right side up. Which face is flipped is determined by the `default_mask_parameters` dict in `klayout_package/python/kqcircuits/layer_config/default_layer_config.py` file. Specifically, the `"chip_trans"` specifies what kind of transformation should be applied to the chips during mask export. For "2b1", it is `pya.DTrans(pya.DVector(10000, 0)) * pya.DTrans().M90,`, which means it shifts the chip for its entire width to the right, then folds it with left edge in place so it gets mirrored (see pya.DTrans().M90 https://www.klayout.de/doc-qt5/code/class_DTrans.html#method15)

But if you have the regions that you want and want to apply the same transformation by hand, just use the transformation `trans = pya.DTrans(pya.DVector(10000, 0)) * pya.DTrans().M90,`. For example if you use `insert_cell` function you can add this transformation as the `trans` argument.

Hope this answers the question and if not let me know the details
